staging/nvidia/charts/nvidia-dcgm-exporter/templates/daemonset.yaml

```diff
@@ -0,0 +1,67 @@
+{{- if .Values.enabled }}
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: nvidia-dcgm-exporter
```

Comment:
Again, should come from the template

Reply:
@joejulian , seems like most of the comments here are about using template for the metadata.name. However, I just looked though most of the charts in the repo (e.g., KUDO, Knative, Istio etc). None of them use the template for the metadata.name.

Could you please explain why that is regarded as best practice? Just for shortening the length of the metadata.name? I would prefer the name to be more explicit instead of requiring people to check the template.

If this is a hard requirement to use the template, could you point me to an existing example in the repo?